### PR TITLE
SD-286 Document new /order endpoint

### PR DESCRIPTION
**Pull request by guidovalente**
I added the new /order endpoint to the documentation, and also polished the readme a bit. Changes are:
- added a TOC
- added an explanation about a 'simple' vs a 'full' integration
- documented new `/order` endpoint

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,49 @@
...
 2. You can use the Tag code to retrieve the QR Tag so that it can be printed and added to the gift's packaging.
 
-## **Retrieving new Tag Codes**
+### Full integration
+
+You will integrate with our API letting StoryDots know when there is a new order, and we will take care of sending a notification to the user so thay they can record their virtual greeting. You will just need to print the tag with the QR code and include it in the gift's package. This integration works as follows:
+
+1. When a new order is placed, you will send a request to the `/order` endpoint informing us of the new purchase. You should perform this request once the purchase _is confirmed_ (this is important since every time you get a Tag code, it will be deducted from your balance).
+2. The response will include the code and the URL to the tag image with the QR code for you to print. You will include the printed tag in the gift's package, and we will take care of sending an email to the user to let them know they can record their virtual greeting.
```
Comment (pwolfus) [inline README.md]: I'd include that the QR image URL is basically the other endpoint we're listing here.

Comment (pwolfus) [reply README.md]: btw, good job on returning this url directly instead of the record url

File: README.md [hunk 2]
```diff
@@ -87,11 +121,66 @@ This API should be integrated in your shopping cart and there are two simple ste
...
 
-  For any questions please send an email to hello@storydots.app
+- **URL**
 
+  _/order_
 
-## **Example code snippets:**
+- **Method:**
+
+  `POST` - assigns a StoryDots code to your new purchase
```
Comment (pwolfus) [inline README.md]: I'd say "order" since we might be confusing "clients" and "purchases" POV

File: README.md [hunk 3]
```diff
@@ -87,11 +121,66 @@ This API should be integrated in your shopping cart and there are two simple ste
...
+
+- **Request Headers**
+
+  **x-api-key:** your private API key that authorizes the request and creates a new order linked to a specific shop
```
Comment (pwolfus) [inline README.md]: lets make this just like the text in /tag so that it's consistent (I'm ok with both simplifying or detailing)

File: README.md [hunk 4]
```diff
@@ -87,11 +121,66 @@ This API should be integrated in your shopping cart and there are two simple ste
...
+  **Content-Type:** application/json
+  
+- **Body params**
```
Comment (pwolfus) [inline README.md]: remove word params

File: README.md [hunk 5]
```diff
@@ -87,11 +121,66 @@ This API should be integrated in your shopping cart and there are two simple ste
...
+
+The params should be sent as a JSON, see the sample call below. These are the params the endpoint will expect:
+  **buyerEmail**: the email address of the buyer to which we will send en email with the recording link
+  **buyerName**: the name of the buyer, helps us customize the experience
```
Comment (pwolfus) [inline README.md]: which help us

File: README.md [hunk 6]
```diff
@@ -87,11 +121,66 @@ This API should be integrated in your shopping cart and there are two simple ste
...
+  **orderId**: your internal ID of the order, we will store this ID in case we need to troubleshoot any issues
```
Comment (pwolfus) [inline README.md]: store this ID to allow you to trace the order if needed

File: README.md [hunk 7]
```diff
@@ -87,11 +121,66 @@ This API should be integrated in your shopping cart and there are two simple ste
...
+  **(optional) buyerPhone** - we will implement SMS/WhatsApp notifications in the near future
+  **(optional) orderDetails** - details about the order being created, which might include products bought, quantity, price, etc. This will enable us to create statistics about your gifts!
```
Comment (pwolfus) [inline README.md:146]: this is freeform json but it would be nice to provide some sort of structure here so that we can easily extract and present it

**Review by pwolfus — CHANGES_REQUESTED** [review]
GJ!

I think it would be awesome in the future to also add a diagram to visually explain the process in the two main scenarios (maybe a flow diagram or activity diagram)

File: README.md (hunk 1 above)
Comment (guidovalente) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (guidovalente) [reply README.md]: agreed, it's changed now

File: README.md (hunk 3 above)
Comment (guidovalente) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (guidovalente) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (guidovalente) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (guidovalente) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (guidovalente) [reply README.md:146]: @pwolfus is that suggestion the text you'd like us to include in the docs or would you like me to produce a sample json with some structure here?

**Pull request merged**